### ISSUE TITLE
Move away from Bintray

### ISSUE THREAD

**Issue by hsablonniere** [CleverCloud/clever-tools#454]
Announcement: https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/

TL;DR: We need to move away from Bintray.

## Details

Our CLI uses Bintray to distribute the binaries for:

* Debian/Ubuntu with `.deb`
* CentOS/Fedora with `.rpm`
* Windows via Chocolatey with `.nupkg`

## TODO

* [x] Find a new location for `.deb`
* [x] Find a new location for `.rpm`
* [x] Find a new location for `.nupkg`
* [x] Modify the build system so we can publish new releases to this new location
* [ ] Automate rpm signature on Jenkins
* [x] Test for deb
* [x] Test for rpm
* [ ] Test for nupkg
* [x] Update install docs
* [x] Publish last version
* [ ] Update related docs page
* [ ] Blogpost maybe

**Comment by hsablonniere** [comment]
Seems like https://help.sonatype.com/repomanager3/formats could work for us.

@ldoguin Can you handle that?

**Comment by hsablonniere** [comment]
@zepag Says we could use Cellar for rpm with this: https://packages.debian.org/fr/jessie/createrepo

We would need a sync step before the publication for S3 (between package and publish).

**Comment by hsablonniere** [comment]
Maybe we could try this for deb: https://manpages.ubuntu.com/manpages/groovy/fr/man1/dpkg-scanpackages.1.html

**Comment by ldoguin** [comment]
@hsablonniere @gaelreyrol Pull request ready for review :)

**Comment by dlecan** [comment]
Any news?

**Comment by fchabouis** [comment]
Hello,
Following the installation instructions for ubuntu, I get the following message :

```
E: The repository 'https://nexus.clever-cloud.com/repository/deb bionic Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
```
Is that something you can fix on your side ? Or am I missing something ?
Thanks

**Comment by hsablonniere** [comment]
Hello people @dlecan @fchabouis

I'm very sorry for this delay.
I'll be able to resume work on this this week.
I'll keep you updated.

**Comment by hsablonniere** [comment]
Hello everyone,

I worked a bit on this issue this week.

* We started the adaptation of `scripts/job-publish-bintray.js` into `scripts/job-publish-nexus.js`.
* It works for stable, we need to make it work for beta.
* We decided that we would create 2 repos for each type (deb, rpm, nupkg) to get stable and beta.
* Nexus's lack of APIs to publish is really annoying, we need to do a custom HTTP request and it's different for each type (deb, rpm, nupkg).
* deb is almost ready, we still need to replace `bionic` into `stable` and update the signature key.
* We also need to remove the beta versions from the stable repos.
* Nexus doesn't do rpm signing, we're looking into another solution with @zepag.
* We haven't tested much nupkg yet.

This task will be resumed on june 15.

**Comment by hsablonniere** [comment]
An idea we need to try to sign rpm:

* Try to run a docker with centos and rpm stuffs to sign and make repo from the jenkins agent (with jnlp) that runs the current jenkins file with all the stages
  * We may need to add the `CC_MOUNT_DOCKER_SOCKET` env var to enable docker in docker in the agent

**Comment by blalasaadri** [comment]
Any update on this? We use an Ubuntu Docker container in our build process to deploy our applications to Clever Cloud. As long as the version of clever-tools isn't updated beyond 2.8.0, we can continue using our current image. But once there is an update, this would have to be fixed for us to use the newest version.

**Comment by hsablonniere** [comment]
Mostly fixed in https://github.com/CleverCloud/clever-tools/pull/504

We still need to work on automatic signature for `.rpm`.

@dlecan @fchabouis @blalasaadri This should be fixed now :tada: 

The `README` should be up to date with correct install docs.
If you have any problems or questions, let us know through the support in the console (or in an issue here).
Thanks for your patience.
